crypto/mem.c

```diff
@@ -112,7 +115,7 @@ static void parseit(void)
     /* Get the count (atol will stop at the @ if there), and percentage */
     md_count = atol(md_failstring);
     atsign = strchr(md_failstring, '@');
-    md_fail_percent = atsign == NULL ? 0 : atoi(atsign + 1);
+    md_fail_percent = atsign == NULL ? 0 : (int)(atof(atsign + 1) * 100 + 0.5);
```

Comment:
(1) This is no longer fail "percent"
(2) `atof()` (and indeed `atoi()`) are generally considered "do not use" by secure coding standards.

For `atoi()` I'd generally prefer something like OpenBSD's (sadly non-standard) `strtonum()`, but for `atof()` I think `strtod()` is pretty much all there is

Reply:
it still means a percentage, just that 0..100% is now encoded as 0..10,000
see below change in `shouldfail()`

Reply:
I don't see `atoi()` or `atof()` as being a problem here.  The error case corresponds to no allocation failures, so it seems unlikely to lead to any problems.